conda_build/__init__.py

```diff
@@ -15,3 +16,17 @@
     "render",
     "skeleton",
 ]
+
+# Skip context logic for doc generation since we don't install all dependencies in the CI doc build environment,
+# see .readthedocs.yml file
+try:
+    import os
+
+    from conda.base.context import reset_context
+
+    # Disallow softlinks. This avoids a lot of dumb issues, at the potential cost of disk space.
+    os.environ["CONDA_ALLOW_SOFTLINKS"] = "false"
+    reset_context()
+
+except ImportError:
+    pass
```

Comment:
This code will be run on every import of `conda_build` and seems like an overreach to fix a documentation rendering issue.

I would suggest to either
- make this conditional [on the default Read The Doc environment variables](https://docs.readthedocs.io/en/stable/reference/environment-variables.html) (messy), or
- move this into a custom Sphinx extension ([how-to](https://www.sphinx-doc.org/en/master/development/tutorials/helloworld.html), basically a module with a `setup(app)` function) (clean).

Reply:
This is supposed to run for ALL imports of `conda_build`, it's only for docs rendering that we don't need this

Reply:
Gotcha, I'd appreciate if we'd not have to handle the special case for the docs here though. Ultimately it's not a huge deal though.